### ISSUE TITLE
compute ctc target failed in layer training.

### ISSUE THREAD

**Issue by Ahmedsabry79** [tesseract-ocr/tesseract#2934]
### Environment

* **Tesseract Version**: 4.00
* **Commit Number**: 
* **Platform**: Kububtu 18.04

### Current Behavior:
I am trying to make a layer training for the ara.traineddata for arabic language so that it can detect some english words if found within an arabic sentence, as well as hindi digits characters.
i prepared the data in a text file, and got this error.

I made a very simple text file to examine this error, the first text file (txt1.txt) did not show the error and the training worked successfully, the second (txt2.txt, which includes a single additional repeated line) returned this error.
Thanks in advance.
[txt1.txt](https://github.com/tesseract-ocr/tesseract/files/4380434/txt1.txt)
[txt2.txt](https://github.com/tesseract-ocr/tesseract/files/4380435/txt2.txt)

**Comment by Shreeshrii** [comment]
Please share the generated box/tif pairs and lstmf files for the above, in order to try and replicate and confirm the error.

**Comment by Ahmedsabry79** [comment]
here are the lstm and box tif files for text1.txt
[ara.Arial.exp0.lstmf.zip](https://github.com/tesseract-ocr/tesseract/files/4385437/ara.Arial.exp0.lstmf.zip)
[ara.Arial.exp0.box.zip](https://github.com/tesseract-ocr/tesseract/files/4385439/ara.Arial.exp0.box.zip)

files of text2.txt
[ara_txt2.Arial.exp0.lstmf.zip](https://github.com/tesseract-ocr/tesseract/files/4385457/ara_txt2.Arial.exp0.lstmf.zip)
[ara_txt2-2020-03-26.Fek.zip](https://github.com/tesseract-ocr/tesseract/files/4385462/ara_txt2-2020-03-26.Fek.zip)

**Comment by Ahmedsabry79** [comment]
I also noticed something strange while training, before training phase starts, we should see:
Loaded 2/2 lines (1-2) of document ../tesstutorial/trainlayer/ara.Arial.exp0.lstmf for text1.txt and 
Loaded 3/3 lines (1-3) of document ../tesstutorial/trainlayer/ara.Arial.exp0.lstmf for text2.txt

but in both cases i see the following:
Loaded 33/33 lines (1-33) of document ../tesstutorial/trainlayer/ara.Arial.exp0.lstmf
it seems that the model is reading the two or three lines as 33 lines for some reason.

**Comment by Shreeshrii** [comment]
Take a look at the box files. Tabs Mark end of lines. (I haven't checked
your files yet, answering from my phone).

There is already an open issue with text2image not generating box files
correctly for RTL.

**Comment by Ahmedsabry79** [comment]
I noticed that when removing the English words error disappear, also i tried to write only Arabic sentences and then English sentences without mixing them as in text2.txt files, but this gave the same error. I am not sure whether the problem is with the left to right and the right to left differences in Arabic and English languages or not.
I looked at the box files, i do not know whether the lines starting by the >> sign are making this problem or not.

**Comment by Shreeshrii** [comment]
I opened the box file in notepad++ and serached for \t (tab character) and there are 32 of them. This is the same issue as reported in https://github.com/tesseract-ocr/tesseract/issues/2655 (still open).

**Comment by Shreeshrii** [comment]
[lstmf2img.zip](https://github.com/tesseract-ocr/tesseract/files/4386101/lstmf2img.zip)

Please the attached zip file. It unpacks the lstmf files you sent using an experimental feature by @stweil.  As you can see the lstmf files are erroneous and hence the errors in training.

**Comment by Ahmedsabry79** [comment]
yes i see, Thanks for your reply, hope this error will be fixed soon.

**Comment by Shreeshrii** [comment]
Unlikely for the error to be fixed soon. We don't have any RTL experts to help fix this.

You can try a two step workaround. Create single line images using text2image but do not use their box files. Instead use the python script from tesstrain repo to create wordstrbox files.

See 
https://github.com/Shreeshrii/tesstrain-ckb/blob/master/2-txt2img.sh
and
https://github.com/Shreeshrii/tesstrain-ckb/blob/master/3-img2lstmf.sh

Basically main commands are:

```
OMP_THREAD_LIMIT=1 text2image --fonts_dir="$unicodefontdir"  --strip_unrenderable_words=false  --ptsize=12  --resolution=300  --xsize=2550  --ysize=300  --leading=32 --margin=100 --exposure=$exp  --font="$fontname" --text="$linetext"  --outputbase="$prefix"/${fontname// /_}.$cnt.exp$exp   --degrade_image=false

rm "$prefix"/${fontname// /_}.$cnt.exp$exp.box
cp "$linetext"  "$prefix"/${fontname// /_}.$cnt.exp$exp.gt.txt
```


```
python3 $SCRIPTPATH/generate_wordstr_box.py -t "${my_file%.*}".gt.txt  -i  "${my_file}" --rtl > "${my_file%.*}".box
OMP_THREAD_LIMIT=1  tesseract "${my_file}" "${my_file%.*}" -l ara --psm 13 --dpi 200  lstm.train    1>/dev/null 2>&1
```

**Comment by Shreeshrii** [comment]
Use this new version of the python script:

https://github.com/Shreeshrii/tesstrain-ckbImpact/blob/master/generate_wordstr_box.py

**Comment by Ahmedsabry79** [comment]
would this affect the ability of the model to extract the text at inference time from a whole page with multiple lines?
